### PR TITLE
protobuf: Update yaml-cpp parser to handle edge case

### PR DESCRIPTION
**Pull request by fishcakez**
Commit Message: protobuf: Update yaml-cpp parser to handle edge case
Additional Description: Bring in an out of memory fix from https://github.com/jbeder/yaml-cpp/pull/865
Risk Level: low
Testing: //test/common/protobuf:utility_test
Docs Changes: n/a
Release Notes: n/a

Signed-off-by: James Fish <jfish@pinterest.com>

### CODE REVIEW

**Comment by fishcakez** [comment]
cc @rgs1

**Comment by sc0ttbeardsley** [comment]
@fishcakez is this needed now that the change has landed upstream?

**Comment by fishcakez** [comment]
> @fishcakez is this needed now that the change has landed upstream?

Instead we need to update the dependency, rather than apply a local patch. Will update this PR.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -263,9 +263,9 @@ DEPENDENCY_REPOSITORIES = dict(
         cpe = "cpe:2.3:a:gnu:zlib:*",
     ),
     com_github_jbeder_yaml_cpp = dict(
-        sha256 = "77ea1b90b3718aa0c324207cb29418f5bced2354c2e483a9523d98c3460af1ed",
-        strip_prefix = "yaml-cpp-yaml-cpp-0.6.3",
-        urls = ["https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.3.tar.gz"],
+        sha256 = "17ffa6320c33de65beec33921c9334dee65751c8a4b797ba5517e844062b98f1",
```
Comment (asraa) [inline bazel/repository_locations.bzl:266]: Thanks! just a drive-by can you add a comment with the date of this commit?

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -263,9 +263,9 @@ DEPENDENCY_REPOSITORIES = dict(
...
+        strip_prefix = "yaml-cpp-6701275f1910bf63631528dfd9df9c3ac787365b",
+        urls = ["https://github.com/jbeder/yaml-cpp/archive/6701275f1910bf63631528dfd9df9c3ac787365b.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl:268]: Please add a date comment for this comment after this line to be consistent with other non-release deps
`# 2020-05-25`

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -263,9 +263,9 @@ DEPENDENCY_REPOSITORIES = dict(
         cpe = "cpe:2.3:a:gnu:zlib:*",
     ),
     com_github_jbeder_yaml_cpp = dict(
-        sha256 = "77ea1b90b3718aa0c324207cb29418f5bced2354c2e483a9523d98c3460af1ed",
-        strip_prefix = "yaml-cpp-yaml-cpp-0.6.3",
-        urls = ["https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.3.tar.gz"],
+        sha256 = "17ffa6320c33de65beec33921c9334dee65751c8a4b797ba5517e844062b98f1",
+        strip_prefix = "yaml-cpp-6701275f1910bf63631528dfd9df9c3ac787365b",
```
Comment (moderation) [inline bazel/repository_locations.bzl:267]: Please add a date comment for this comment after this line to be consistent with other non-release deps
`# 2020-05-25`

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (fishcakez) [reply bazel/repository_locations.bzl:266]: Fixed :)

File: bazel/repository_locations.bzl (hunk 3 above)
Comment (fishcakez) [reply bazel/repository_locations.bzl:267]: Fixed :)

**Comment by mattklein123** [comment]
Windows CI failure looks legit. cc @wrowe

**Comment by wrowe** [comment]
> Windows CI failure looks legit. cc @wrowe

Unfortunately, https://dev.azure.com/cncf/envoy/_build/results?buildId=40739&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=168f295b-0553-5364-35f7-923225ecd8b3&l=103 doesn't give us enough indication of the issue. @sunjayBhatia and I are each working on different projects Monday, but if we can see this in any higher resolution locally, we'll provide more feedback. Tuesday, for sure.

**Comment by fishcakez** [comment]
Since yaml-cpp has bazel support built in, I might try my luck later today to move from foreign_cc -> yamlccp's own build rules. I don't have access to windows to debug locally.

**Comment by wrowe** [comment]
Sorry, we had promised Tuesday... breakage of test on master eclipsed this project, but we will take a look ourselves tomorrow to see if we can get some more detailed insights.

**Comment by wrowe** [comment]
cc @sunjayBhatia
